bundles/org.openhab.binding.mybmw/src/main/java/org/openhab/binding/mybmw/internal/handler/backend/MyBMWHttpProxy.java

```diff
@@ -0,0 +1,353 @@
+    private final String remoteStatusUrl;
+
+    public MyBMWHttpProxy(HttpClientFactory httpClientFactory, MyBMWBridgeConfiguration bridgeConfiguration) {
+        logger.trace("MyBMWHttpProxy - initialize");
+        httpClient = httpClientFactory.getCommonHttpClient();
+
+        myBMWTokenHandler = new MyBMWTokenController(bridgeConfiguration, httpClient);
+
+        this.bridgeConfiguration = bridgeConfiguration;
+
+        vehicleUrl = "https://" + BimmerConstants.EADRAX_SERVER_MAP.get(bridgeConfiguration.region)
+                + BimmerConstants.API_VEHICLES;
+
+        vehicleStateUrl = vehicleUrl + "/state";
+
+        remoteCommandUrl = "https://" + BimmerConstants.EADRAX_SERVER_MAP.get(bridgeConfiguration.region)
+                + BimmerConstants.API_REMOTE_SERVICE_BASE_URL;
+        remoteStatusUrl = remoteCommandUrl + "eventStatus";
+        logger.trace("MyBMWHttpProxy - ready");
+    }
+
+    @Override
+    public void setBridgeConfiguration(MyBMWBridgeConfiguration bridgeConfiguration) {
+        this.bridgeConfiguration = bridgeConfiguration;
+    }
+
+    public List<@NonNull Vehicle> requestVehicles() throws NetworkException {
+        List<@NonNull Vehicle> vehicles = new ArrayList<>();
+        List<@NonNull VehicleBase> vehiclesBase = requestVehiclesBase();
+
+        for (VehicleBase vehicleBase : vehiclesBase) {
+            VehicleStateContainer vehicleState = requestVehicleState(vehicleBase.getVin(),
+                    vehicleBase.getAttributes().getBrand());
+
+            Vehicle vehicle = new Vehicle();
+            vehicle.setVehicleBase(vehicleBase);
+            vehicle.setVehicleState(vehicleState);
+            vehicles.add(vehicle);
+        }
+
+        return vehicles;
+    }
+
+    /**
+     * request all vehicles for one specific brand and their state
+     *
+     * @param brand
+     */
+    public List<VehicleBase> requestVehiclesBase(String brand) throws NetworkException {
+        String vehicleResponseString = requestVehiclesBaseJson(brand);
+        return JsonStringDeserializer.getVehicleBaseList(vehicleResponseString);
+    }
+
+    public String requestVehiclesBaseJson(String brand) throws NetworkException {
+        byte[] vehicleResponse = get(vehicleUrl, brand, null, HTTPConstants.CONTENT_TYPE_JSON);
+        String vehicleResponseString = new String(vehicleResponse, Charset.defaultCharset());
+        return vehicleResponseString;
+    }
+
+    /**
+     * request vehicles for all possible brands
+     *
+     * @param callback
+     */
+    public List<VehicleBase> requestVehiclesBase() throws NetworkException {
+        List<VehicleBase> vehicles = new ArrayList<>();
+
+        for (String brand : BimmerConstants.REQUESTED_BRANDS) {
+            try {
+                vehicles.addAll(requestVehiclesBase(brand));
+
+                Thread.sleep(10000);
+            } catch (Exception e) {
+                logger.error("error retrieving the base vehicles {}", e.getMessage());
```

Comment:
Will the binding stop working in this case? Otherwise log level should be decreased. See https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
you're right, I reduced it to warn level